drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/auditlog/AuditLogEntryCellHelperTest.java

```diff
@@ -0,0 +1,619 @@
+import org.drools.workbench.models.guided.dtable.shared.model.LimitedEntryActionSetFieldCol52;
+import org.drools.workbench.models.guided.dtable.shared.model.LimitedEntryConditionCol52;
+import org.drools.workbench.models.guided.dtable.shared.model.MetadataCol52;
+import org.drools.workbench.models.guided.dtable.shared.model.Pattern52;
+import org.drools.workbench.models.guided.dtable.shared.model.WorkItemColumnParameterValueDiffImpl;
+import org.drools.workbench.screens.guided.dtable.client.resources.i18n.GuidedDecisionTableConstants;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class AuditLogEntryCellHelperTest {
+
+    @Mock
+    private DateTimeFormat format;
+    private String labelClass = "mockCssClassForLabel";
+    private AuditLogEntryCellHelper helper;
+
+    @Before
+    public void setUp() throws NoSuchFieldException, IllegalArgumentException, IllegalAccessException {
+        when(format.format(any(Date.class))).thenReturn("mockDateString");
+        // This also mock AuditLogEntryCellHelper.Template, which then for each method call returns "<mathod name>(<call arguments>)
+        // instead of the "templated" String. Calling SafeHtml.asString() then gives us something like:
+        // commentHeader(<new header>)updatedFields(updatedField(<field description>, <old value>, <new value>)...)
+        // From that we can assert that all diffs were properly used and displayed.
+        helper = new AuditLogEntryCellHelper(format, labelClass, "mockCssClassForValue");
+    }
+
+    @Test
+    public void convertValueToString() {
+        assertEquals(null, helper.convertValueToString(null));
+        assertEquals("true", helper.convertValueToString(true));
+        assertEquals("false", helper.convertValueToString(false));
+        Date date = new Date();
+        assertEquals(format.format(date), helper.convertValueToString(date));
+        assertEquals(BigDecimal.ONE.toPlainString(), helper.convertValueToString(BigDecimal.ONE));
+        assertEquals(BigInteger.ONE.toString(), helper.convertValueToString(BigInteger.ONE));
+        Byte b = Byte.MAX_VALUE;
+        assertEquals(b.toString(), helper.convertValueToString(b));
+        Double d = 123.456;
+        assertEquals(d.toString(), helper.convertValueToString(d));
+        Float f = new Float(123.456);
+        assertEquals(f.toString(), helper.convertValueToString(f));
+        Integer i = 123;
+        assertEquals(i.toString(), helper.convertValueToString(i));
+        Long l = new Long(123);
+        assertEquals(l.toString(), helper.convertValueToString(l));
+        Short s = 123;
+        assertEquals(s.toString(), helper.convertValueToString(s));
+        assertEquals("surprise!", helper.convertValueToString("surprise!"));
+        assertEquals(this.toString(), helper.convertValueToString(this));
+    }
+
+    @Override
+    public String toString() {
```

Comment:
It seemed to be a nice/easy way to test that `helper.convertValueToString(..)` doesn't blow up on unexpected types (see the original code for the 'expected' types). I can remove the check and the `toString()` though, no problem.